### PR TITLE
Removing Settings::General.email_addresses

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
In #13853 I replaced granular emails from `Settings::General.email_addresses` hash with the default one, but `Settings::General.email_addresses` remained to be a hash.
In this pr `Settings::General.email_addresses` was replaced with `Forem.email`
I also removed the redundant `DataUpdateScript` (db record and file) related to email addresses.

## Related Tickets & Documents
#13853
#13628

## Added tests?

- [ ] Yes
- [x] No, and this is why: changed the related tests where needed
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: no major changes in the logic/behavior

### CODE REVIEW

File: lib/data_update_scripts/20210604104735_remove_email_addresses_scripts.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module DataUpdateScripts
+  class RemoveEmailAddressesScripts
+    def run
+      DataUpdateScript.delete_by(file_name: "20210509105151_remove_unused_site_config_emails")
```
Comment (rhymes) [inline lib/data_update_scripts/20210604104735_remove_email_addresses_scripts.rb:4]: 🔥 

We should also remove the line from `Settings::General`'s DB table:

```ruby
Settings::General.delete_by(var: :email_addresses)
```

**Review by rhymes — APPROVED** [review]
Well done, left a note about cleaning the settings table as well!

We came full circle with the emails, there used to be one, then many and now back to one :D

File: app/views/admin/settings/show.html.erb [hunk 2]
```diff
@@ -602,12 +601,11 @@
                 <div class="crayons-field__description">
                   Default (set via <span class="ff-monospace">default_email</span> environment variable)
                 </div>
-                <input type="text" value="<%= ApplicationConfig["DEFAULT_EMAIL"] %>" class="form-control" readonly>
+                <input type="text" value="<%= ForemInstance.email %>" class="form-control" readonly>
```
[the remark this answers is not in the document]
Comment (lightalloy) [reply app/views/admin/settings/show.html.erb]: Thanks for the comment. I changed the `id` and the `label` on this page.

**Review by jgaskins — APPROVED** [review]
Thank you for taking care of that label! 💯 🎉

**Review by rhymes — APPROVED** [review]
LGTM!

File: app/views/admin/settings/show.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (aitchiss) [reply app/views/admin/settings/show.html.erb]: nice catch! the update looks good to me 🚀

**Pull request merged**